client/src/app.js

```diff
@@ -35,6 +36,7 @@ class App extends Component {
           <SignupPage path="signup" />
           <InactivePage path="inactive" />
           <NotFoundPage default />
+          <PrivateRoute as={UserProfilePage} path="userprofile" />
```

Comment:
```suggestion
          <PrivateRoute as={UserProfilePage} path=":authorId" />
```

authorId: string

1. Local user
the id (primary key) of the user

2. Foreign user
it will follow such convention
```js
let username = encodeURIComponent(`http(s)://${foreign_hostname}/authors/<id>`)
```

The `UserProfilePage` should be able to handle both cases, and fetch extra author information as needed.

However, you may assume we do not have any foreign user for now.

Reply:
Actually, we only show friend list for login user himself, and we don't show on other user profile page.